site/assets/js/cookie-consent.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Cookie Consent Banner JavaScript
+ * Copyright (C) CVAT.ai Corporation
+ * SPDX-License-Identifier: MIT
+ *
+ * Note: Banner visibility is controlled by Hugo production mode.
+ * This script only loads in production environments.
+ */
+
+(function() {
+    'use strict';
+
+    /**
+     * Simple check for localhost/development to prevent analytics pollution
+     * @returns {boolean} True if running on localhost or with custom port
+     */
+    function isLocalhost() {
+        return location.hostname.includes('localhost') ||
+               location.hostname === '127.0.0.1' ||
+               location.port !== '';
+    }
+
+    /**
+     * Inject Google Analytics script into the page
+     */
+    function injectGAScript() {
+        // Defense in depth: prevent analytics on localhost even in production builds
+        if (isLocalhost()) {
+            console.log('Analytics disabled for localhost');
+            return;
+        }
+
+        // Check if GA tracking ID is available
+        if (!window.GOOGLE_TAG_ID) {
+            console.log('GA tracking ID not configured');
+            return;
+        }
+
+        if (document.getElementById('ga-script')) return;
+
+        // Re-enable Google Analytics tracking if it was previously disabled
+        if (window.GOOGLE_TAG_ID) {
```

Comment:
This condition is redundant - you have an earlier `if` that returns from the function if the tag ID is not set.